test/functionalTest/cases/3190_notify_only_attributes_that_change/notify_only_attributes_that_change_with_cond_with_notify.test

```diff
@@ -0,0 +1,483 @@
+# Copyright 2019 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Notify only attributes that change for NGSIv2 POST /v2/entities, POST /v2/subscriptions, Dump and reset accumulator, PATCH /v2/entities/Room1/attrs, Dump and reset accumulator, PATCH /v2/entities/Room1/attrs, Dump and reset accumulator, PATCH /v2/entities/Room1/attrs, Dump and reset accumulator, PATCH /v2/entities/Room1/attrs, Dump and reset accumulator.
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print localhost $LISTENER_PORT
+
+--SHELL--
+
+#
+# 01. POST /v2/entities, create Room1 entity with A, B, C, and D attributes.
+# 02. POST /v2/subscriptions, to create subscription for Room1 with cond attrs: [A, B, C] notify attrs: [B, C, D], onlyChangedAttrs: true.
+# 03. Dump and reset accumulator, see B, C and D (as this is initial notification).
+# 04. PATCH /v2/entities/Room1/attrs, update the value of A and B attribute.
+# 05. Dump and reset accumulator, see B.
+# 06. PATCH /v2/entities/Room1/attrs, update the value of B and C attribute.
+# 07. Dump and reset accumulator, see B and C.
+# 08. PATCH /v2/entities/Room1/attrs, update the value of C and D attribute.
+# 09. Dump and reset accumulator, see C.
```

Comment:
I think this should be:

"Dump and reset accumulator, see C **and D**"

My rationale is:

* C and D are being changed. C is in the attrs array ([A, B, C]) so the notification is triggered
* Notification is composed by [B, C, D] but only those that changes changed (that is: C and D). So resulting notification is [C, D]

Similarly, D should be included in step 11.

Reply:
fixed in c089631